src/main/java/org/commcare/formplayer/services/FormDefinitionService.java

```diff
@@ -59,4 +60,31 @@ public SerializableFormDefinition getOrCreateFormDefinition(
         formDefinition.setFormDef(formDef);
         return formDefinition;
     }
+
+    /**
+     * This method uses appId, formXmlns, formVersion, and sessionId to cache deserizlied FormDefs
+     * Deserializing a serialized FormDef object is costly, and this avoids doing so in between requests
+     * within the same session
+     *
+     * @param appId application id
+     * @param formXmlns xmlns for specific form
+     * @param formVersion version for specific form
+     * @param sessionId form session id
+     * @param serializedFormDef serialized FormDef object
+     * @return deserialized FormDef object
+     */
+    @Cacheable(key = "{#appId, #formXmlns, #formVersion, #sessionId}")
+    public FormDef getFormDef(String appId,
+            String formXmlns,
+            String formVersion,
```

Comment:
The serializable form session ID is unique across all sessions, so we should be able to just cache on that right?

Reply:
See 0cefa6fb921734998a392837e4f06c12aa23126f